### PR TITLE
Addresses review comments of previously merged IMap#flush related PR

### PR DESCRIPTION
**Pull request by ahmetmircik**
This PR aims to address the review comments of previously merged PR, for example, https://github.com/hazelcast/hazelcast/pull/7383#issuecomment-175582761

- Addresses potential latency problem of IMap#flush call when there are subsequent IMap#flush calls.
   Introduces `sequence` to find when to return answer upon call of IMap#flush.
- Uses wait/notify service while waiting end of IMap#flush
- Makes some code simplifications
- Added `EvictBackupOperation` as a backup of `EvictOperation`

closes https://github.com/hazelcast/hazelcast/issues/7182
closes https://github.com/hazelcast/hazelcast-enterprise/issues/735

### CODE REVIEW

**Comment by ahmetmircik** [comment]
run-lab-run

**Comment by ahmetmircik** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/writebehind/BoundedWriteBehindQueue.java [hunk 1]
```diff
@@ -149,26 +154,9 @@ public void clear() {
...
-    /**
-     * Adds some number of elements to the supplied collection starting from the head of this queue.
-     *
-     * @param numberOfElements number of elements to add.
-     * @param collection       to add filtered elements.
-     */
     @Override
-    public void getFrontByNumber(int numberOfElements, Collection<E> collection) {
-        queue.getFrontByNumber(numberOfElements, collection);
+    public void filter(IPredicate<E> predicate, Collection<E> collection) {
```
Comment (gurbuzali) [inline hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/writebehind/BoundedWriteBehindQueue.java]: can you add some javadoc about the filtering

**Comment by gurbuzali** [comment]
:+1:  with a minor comment

File: hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/writebehind/BoundedWriteBehindQueue.java (hunk 1 above)
Comment (ahmetmircik) [inline hazelcast/src/main/java/com/hazelcast/map/impl/mapstore/writebehind/BoundedWriteBehindQueue.java]: It is already there, on the interface, please see https://github.com/hazelcast/hazelcast/pull/7428/files#diff-79bbf4022aca33f09a86c90f0160993aR98

**Comment by eminn** [comment]
:+1:

**Pull request merged**